record/log_writer.go

```diff
@@ -20,6 +20,7 @@ import (
 )
 
 var walSyncLabels = pprof.Labels("pebble", "wal-sync")
+var errClosed = errors.New("pebble/record: closed LogWriter")
```

Comment:
nit: Depending on the namespace we're operating in (the `record` package), we may want to consider giving this a more descriptive name. Something like `errClosedLogWriter`.